### DISCUSSION TITLE
Setting up elemo locally as mentioned in README.md gives error

### DISCUSSION THREAD

**Discussion by unamdev0** [opcotech/elemo#243] [Q&A] [answered]
Hey Everyone, I'm trying to setup elemo locally as to start contributing. But I'm not able to get the application started, as it's throwing error 
`line 5: CI: unbound variable`

I've made sure all the required dependencies are already installed in my system.

Can someone please look into this and help me in setting up the app.

Thanks!!

<img width="706" alt="Screenshot 2024-08-17 at 5 37 05 PM" src="https://github.com/user-attachments/assets/c5883d19-e90e-46a9-9a3f-1569d3ea7cd9">

UPDATE: I was able to run the app by removing both the If statements in the starting of `generate-dev-config.sh` and `setup.sh` files

**Comment by unamdev0** [comment]
There also seems to be some issue with pre defined permission for demo user,as all the requests are being blocked

<img width="706" alt="Screenshot 2024-08-17 at 8 40 27 PM" src="https://github.com/user-attachments/assets/119d383a-6789-4dc1-a8dc-add8191d72d0">

**Comment by gabor-boros** [comment]
Hey @unamdev0,

Good to have you there and to hear you would like to contribute 😊

Ah, the CI variable unbound issue... I thought I fixed that last time. I'll open a PR to fix that later today/tomorrow.

Regarding the permission issue, can you see anything in the browser's dev console or in the network tab?

Also what browser do you use? I'd love to reproduce it.

**Reply by unamdev0** [reply]
Hi @gabor-boros 

Yes, there's total of 5 consoles logs that are coming, 4 of which are regarding the blocked requests and 1 is regarding webpack, PFA log file. 
This seems like a browser issue and *not a permission issue* because when I'm trying one of these requests in postman, I'm getting empty array as response.

I'm using brave browser(Version 1.68.141 Chromium: 127.0.6533.120)

<img width="646" alt="Screenshot 2024-08-18 at 11 58 49 AM" src="https://github.com/user-attachments/assets/9b6e3516-3994-4343-9e38-9485dc66e348">


<img width="646" alt="Screenshot 2024-08-18 at 11 58 40 AM" src="https://github.com/user-attachments/assets/43d6e71f-ed44-4f1e-9a3b-e592119fab46">

[127.0.0.1-1723962697452.log](https://github.com/user-attachments/files/16647440/127.0.0.1-1723962697452.log)

**Reply by unamdev0** [reply]
Also I'm not sure if this is a bug or intended result,but I'm not able to access namespace,project or document's page 

<img width="777" alt="Screenshot 2024-08-18 at 12 03 17 PM" src="https://github.com/user-attachments/assets/90ae0d47-a47f-417f-ac46-016725ba379c">

**Reply by gabor-boros** [reply, marked as answer]
@unamdev0 So, the CI unbound variable issue is fixed by https://github.com/opcotech/elemo/commit/bf444b6b109a2860683fd6df43912f67ab1a58d8. Regarding the 404 pages, only the following pages are done or **partially** done so far:

- signin
- profile
- settings/organizations
- settings/security
- settings/system

By the way, I'm going to push a half-way-done refactor with a new front-end which has way better bases. Which part would you want to help out with?

Based on your latest screenshot, am I right that you solved the browser issues?

**Reply by gabor-boros** [reply]
FTR: the PR of front-end refactor that I was referring to: https://github.com/opcotech/elemo/pull/244

**Reply by unamdev0** [reply]
@gabor-boros, 

I'd like to start with backend, I'm new to golang and would like to contribute to it.
I'll try to resolve [#149](https://github.com/opcotech/elemo/issues/149) once I'm able to hit APIs from browser

Also I've not been able to resolve the `Blocked by client` error which is coming in browser.


I've checked and CORS seems to be enabled, but I'm still not able to hit any API through browser. 

Can you please suggest some solution which I can try

**Reply by gabor-boros** [reply]
@unamdev0 I have the feeling that brave's "shields" are blocking the request

**Reply by unamdev0** [reply]
Great, removing brave shields works!!! Thanks a ton

I'll try to understand the code in a better way,and then try to resolve #149. Once I'm confident enough, I'll ask you to assign the issue to me.

**Reply by gabor-boros** [reply]
I'm happy that it solved the issue for you! I'm going to mark the comment as the answer if you don't mind.

If you need more information or details regarding that issue, feel free to ping me there 😊

**Reply by unamdev0** [reply]
Sure!! thanks again for helping
